Firestore/core/src/firebase/firestore/core/query.h

```diff
@@ -39,13 +40,12 @@ namespace core {
 using CollectionGroupId = std::shared_ptr<const std::string>;
 
 /**
- * Represents the internal structure of a Firestore Query. Query instances are
- * immutable.
+ * Encapsulates all the query attributes we support in the SDK. It can be run
```

Comment:
One feature that distinguishes Queries vs Targets that isn't well captured in this comment is that `Query` represents the user-facing concept of the query where `Target` represents the query we might execute on the server. The comment says this indirectly now (in its discussion of the LocalStore vs RemoteStore) but making this explicit would make this clearer.